File: 71/71.cs
public class Solution {
    public string SimplifyPath(string path) {
        string[] arr =  path.Split('/');
        // Console.WriteLine("{0}", string.Join("/", arr.Take(2)));
        int len = arr.Length;
        int ptr = 1;
        for (int i = 1; i<len; i++)
        // foreach (string item in arr)
        {
            // Console.WriteLine(arr[i]);           

            // 迭代器会慢一点
            // if (item == "." || item == "") continue;
            // if (item == ".." ) {
            //     if (ptr>1) ptr--; 
            //     continue;
            // }
            // arr[ptr] = item;
            // ptr++;
            if (arr[i].Length <= 2){
                if (arr[i] == "." || arr[i].Length == 0) continue;
                if (arr[i] == ".." ) {
                    if (ptr>1) ptr--; 
                    continue;
                }
            }
            arr[ptr] = arr[i];
            ptr++;
        }
        if (ptr==1) return "/";
        StringBuilder builder = new StringBuilder();
        for (int i = 1; i<ptr; i++)
        {
            builder.Append('/');
            builder.Append(arr[i]);
        }
        return builder.ToString();

        // return string.Join("/", arr.Take(ptr));
    }
}

File: 71/71.java
class Solution {
    public String simplifyPath(String path) {
        String[] arrOfStr = path.split("/", -1);

        ArrayList<String> res = new ArrayList<String>();
        // ArrayList<String> res = new ArrayList<String>(arrOfStr.length);
        
        StringBuilder sb = new StringBuilder();

        // System.out.println(arrOfStr);
        for (String s : arrOfStr){
            // System.out.println(s);
            if (s.equals("") || s.equals(".")) continue;
            if (s.equals("..")) {
                if (res.size() != 0)
                    res.remove(res.size()-1);
                // continue;
            }else{
                res.add(s);
            }
        }
        // String ans = "";
        for (String s : res){
            sb.append('/');
            sb.append(s);
        }
        if (sb.length() == 0) sb.append('/');
        return sb.toString();
    }
}
